bindings/types.go

```diff
@@ -0,0 +1,194 @@
+		"1.0.0": "1.3.0",
+		"1.1.0": "1.9.0",
+		// add more versions here
+	}
+	v, ok := compatibleVersions[flixVersion]
+	if !ok {
+		// default to latest if flix version not configured
+		return "1.9.0"
+	}
+	return v
+}
+
+func getTemplateDataV1_1(flix *v1_1.InteractionTemplate, templateLocation string, isLocal bool) templateData {
+	var msgs v1_1.InteractionTemplateMessages = flix.Data.Messages
+	title := msgs.GetTitle("Request")
+	methodName := strcase.LowerCamelCase(title)
+	description := msgs.GetDescription("")
+	result := simpleParameter{}
+	if flix.Data.Output != nil {
+		o := transformParameters([]v1_1.Parameter{*flix.Data.Output})
+		if len(o) > 0 {
+			result = o[0]
+		}
+	}
+	data := templateData{
+		Version:              flix.FVersion,
+		Parameters:           transformParameters(flix.Data.Parameters),
+		ParametersPrefixName: strcase.UpperCamelCase(title),
+		Output:               result,
+		Title:                methodName,
+		Description:          description,
+		Location:             templateLocation,
+		IsScript:             flix.IsScript(),
+		IsLocalTemplate:      isLocal,
+	}
+	return data
+}
+
+func getTemplateDataV1_0(flix *flixkit.FlowInteractionTemplate, templateLocation string, isLocal bool) templateData {
+	title := flix.Data.Messages.GetTitleValue("Request")
+	methodName := strcase.LowerCamelCase(title)
+	description := flix.GetDescription()
+
+	data := templateData{
+		Version:              flix.FVersion,
+		Parameters:           transformArguments(flix.Data.Arguments),
+		ParametersPrefixName: strcase.UpperCamelCase(title),
+		Title:                methodName,
+		Description:          description,
+		Location:             templateLocation,
+		IsScript:             flix.IsScript(),
+		IsLocalTemplate:      isLocal,
+	}
+	return data
+}
+
+func convertCadenceTypeToJS(cadenceType string) string {
+	// need to determine js type based on fcl supported types
+	// looking at fcl types and how arguments work as parameters
+	// https://github.com/onflow/fcl-js/blob/master/packages/types/src/types.js
+	switch cadenceType {
+	case "Bool":
+		return "boolean"
+	case "Void":
+		return "void"
+	case "Dictionary":
+		return "object"
+	case "Struct":
+		return "object"
+	case "Enum":
+		return "object"
+	default:
+		return "string"
+	}
+}
+
+func parseTemplates(templates []string) (*template.Template, error) {
+	baseTemplate := template.New("base")
+
+	for _, tmplStr := range templates {
+		_, err := baseTemplate.Parse(tmplStr)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return baseTemplate, nil
+}
+
+func transformParameters(args []v1_1.Parameter) []simpleParameter {
+	simpleArgs := []simpleParameter{}
+	if len(args) == 0 {
+		return simpleArgs
+	}
+	sort.Slice(args, func(i, j int) bool {
+		return args[i].Index < args[j].Index
+	})
+
+	for _, arg := range args {
+		isArray, cType, jsType := isArrayParameter(FlixParameter{Name: arg.Label, Type: arg.Type})
+		var msgs v1_1.InteractionTemplateMessages = arg.Messages
```

Comment:
you should just be able to do 
```suggestion
	 msgs := arg.Messages
```
unless im missing something